File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.math.round.overload/vb/midpoint1.vb
' Visual Basic .NET Document
Option Strict On

Module Midpoint1
    Public Sub Example()
        ' <Snippet5>
        Console.WriteLine("{0,-10} {1,-10} {2,-10} {3,-15} {4,-15}", "Value", "Default",
                        "ToEven", "AwayFromZero", "ToZero")
        For value As Decimal = 12D To 13D Step 0.1D
            Console.WriteLine("{0,-10} {1,-10} {2,-10} {3,-15} {4,-15}",
                           value, Math.Round(value),
                           Math.Round(value, MidpointRounding.ToEven),
                           Math.Round(value, MidpointRounding.AwayFromZero),
                           Math.Round(value, MidpointRounding.ToZero))
        Next

        ' The example displays the following output:
        '       Value      Default    ToEven     AwayFromZero     ToZero
        '       12         12         12         12               12
        '       12.1       12         12         12               12
        '       12.2       12         12         12               12
        '       12.3       12         12         12               12
        '       12.4       12         12         12               12
        '       12.5       12         12         13               12
        '       12.6       13         13         13               12
        '       12.7       13         13         13               12
        '       12.8       13         13         13               12
        '       12.9       13         13         13               12
        '       13.0       13         13         13               13
        ' </Snippet5>
    End Sub
End Module

File: snippets/fsharp/System/Decimal/Round/midpoint1.fs
module midpoint1

open System

// <Snippet5>
printfn $"""{"Value",-10} {"Default",-10} {"ToEven",-10} {"AwayFromZero",-15} {"ToZero",-15}"""
for value in 12m .. 0.1m .. 13m do
    printfn "%-10O %-10O %-10O %-15O %-15O" 
        value
        (Math.Round value)
        (Math.Round(value, MidpointRounding.ToEven))
        (Math.Round(value, MidpointRounding.AwayFromZero))
        (Math.Round(value, MidpointRounding.ToZero))

// The example displays the following output:
//       Value      Default    ToEven     AwayFromZero    ToZero
//       12.0       12         12         12              12
//       12.1       12         12         12              12
//       12.2       12         12         12              12
//       12.3       12         12         12              12
//       12.4       12         12         12              12
//       12.5       12         12         13              12
//       12.6       13         13         13              12
//       12.7       13         13         13              12
//       12.8       13         13         13              12
//       12.9       13         13         13              12
//       13.0       13         13         13              13
// </Snippet5> 